static-analysis-plugin/plugin/src/test/groovy/com/novoda/test/TestProject.groovy

```diff
@@ -0,0 +1,111 @@
+package com.novoda.test
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+
+abstract class TestProject {
+    private static final Closure<String> EXTENSION_TEMPLATE = { TestProject project ->
+        """
+staticAnalysis {
+    ${formatPenalty(project)}
+}
+"""
+    }
+
+    private final File projectDir
+    private final GradleRunner gradleRunner
+    private final Closure<String> template
+    Map<String, List<File>> sourceSets = [main: []]
+    String penalty
+
+    TestProject(Closure<String> template) {
+        this.template = template
+        this.projectDir = createProjectDir("${System.currentTimeMillis()}")
+        this.gradleRunner = GradleRunner.create()
+                .withProjectDir(projectDir)
+                .withDebug(true)
+                .withPluginClasspath()
+                .forwardStdOutput(new OutputStreamWriter(System.out))
+                .forwardStdError(new OutputStreamWriter(System.out))
+    }
+
+    private static File createProjectDir(String path) {
+        File dir = new File(Fixtures.BUILD_DIR, "test-projects/$path")
```

Comment:
every `TestProject` will create a project folder in the build folder of the `plugin` project.